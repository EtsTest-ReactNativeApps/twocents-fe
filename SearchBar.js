import React from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';

export class SearchBar extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <TextInput placeHolder='hey' style={styles.input}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    width: '100%'
  },
  input: {
    color: '#EE933F',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    width: '80%',
    height: 30,
    borderRadius: 20,
  }
});