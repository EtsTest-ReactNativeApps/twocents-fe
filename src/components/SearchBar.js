import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';

class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.state ={
      input: ''
    }
  }
  render() {
    const handleChange = enteredText => {
      this.setState({input: enteredText})
    }  
    const handleSubmit = () => {
      console.log(this.state.input)
      this.setState({input: ''})
    }
    return (
      <View style={styles.container}>
        <TextInput 
          style={styles.input}
          value={this.state.input}
          onChangeText={handleChange}
          onSubmitEditing={handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    height: 35,
    width: '84%',
    borderRadius: 20,
  },
  input: {
    width: '90%',
    color: '#FFF',
  }
});

export default SearchBar;