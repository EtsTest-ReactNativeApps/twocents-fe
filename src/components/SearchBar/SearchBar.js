import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';

export class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.state ={
      input: '',
      searchValues: []
    }
  }
  render() {
    const handleChange = enteredText => {
      this.setState({input: enteredText})
    }  
    const handleSubmit = () => {
      this.setState({input: ''})
      getSearchResults()
    }

    const getSearchResults = () => {
      let searchRecs = this.props.allRecommendations.filter(rec => rec.name.includes(this.state.input))
      console.log(searchRecs)
    }
    return (
      <View style={styles.container}>
        <TextInput 
          placeholder='Enter saved location'
          style={styles.input}
          value={this.state.input}
          onChangeText={handleChange}
          onSubmitEditing={handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    height: 35,
    width: '84%',
    borderRadius: 20,
  },
  input: {
    width: '90%',
    color: '#FFF',
  }
});

const mapStateToProps = state => ({
  allRecommendations: state.allRecommendations
})

export default connect(mapStateToProps)(SearchBar) ;