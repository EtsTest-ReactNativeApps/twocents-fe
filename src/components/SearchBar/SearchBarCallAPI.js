import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';

export class SearchBarCallAPI extends Component {
  constructor(props) {
    super(props)
    this.state ={
      name: '', 
      address: 'Current Location',
      error: ''
    }
  }

  handleChangeName = enteredText => {
    this.setState({name: enteredText})
  } 
  
  handleChangeAddress = enteredText => {
    this.setState({address: enteredText})
  }

  handleSubmit = () => {
    if(!this.state.name || !this.state.address) {
      this.setState({error: 'Missing required field'})
    } else {
      if(this.state.address === 'Current Location') {
       console.log(this.props.userLocation)
      } else {
       console.log(this.state.address)
      }
    }
    this.setState({name: '', address: 'Current Location'})
  }

  render() {
    return (
      <View style={styles.container}>
          <TextInput 
            placeholder='Enter Recommended Location'
            style={styles.input}
            value={this.state.name}
            onChangeText={this.handleChangeName}
            onSubmitEditing={this.handleSubmit}
          />
          <TextInput 
            placeholder='Enter City and State'
            style={styles.input}
            value={this.state.address}
            onChangeText={this.handleChangeAddress}
            onSubmitEditing={this.handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(204, 192, 221, 0.4)',
    height: 70,
    width: '84%',
    borderRadius: 20,
  },
  input: {
    width: '90%',
    color: '#FFF',
  }
});

const mapStateToProps = state => ({
  userLocation: state.userLocation
})

export default connect(mapStateToProps)(SearchBarCallAPI);