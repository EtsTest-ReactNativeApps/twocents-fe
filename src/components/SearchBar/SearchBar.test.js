import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from './SearchBar';

describe('SearchBar', () => {

  it('should match to snapshot', () => {
    const component = shallow(<SearchBar/>)
    expect(component).toMatchSnapshot()
  });
  
});