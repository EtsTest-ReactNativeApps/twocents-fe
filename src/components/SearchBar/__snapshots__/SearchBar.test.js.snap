// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(204, 192, 221, 0.4)",
      "borderRadius": 20,
      "height": 35,
      "justifyContent": "space-around",
      "width": "84%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Enter saved location"
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "color": "#FFF",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
